blackbird_python/blackbird/tests/test_utils.py

```diff
@@ -0,0 +1,388 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the utils module"""
+from textwrap import dedent
+from collections import OrderedDict
+
+import pytest
+
+import numpy as np
+import sympy as sym
+
+from blackbird import loads
+from blackbird.listener import RegRefTransform
+from blackbird.utils import to_DiGraph, match_template, TemplateError
+
+
+class TestDiGraph:
+    """Test conversion of programs to DAGs"""
+
+    def test_no_args(self):
+        """Test case where operations have no arguments"""
+        program = loads(dedent(
+            """\
+            name prog
+            version 1.1

```

Comment:
throughout: again, it feels strange to put specific versions in here. If they must be there, could we not put fake version numbers? `version x.y`? `version -1.0`? `version dev.test`?

Reply:
The only reason version is included is because I've made it a required statement. Should I replace it with something fake for the tests, like `version 0.0`?

Reply:
Updated all version numbers to 0.0.